test/test_common.py

```diff
@@ -88,6 +90,40 @@ def test_json_serialization(self, tmpdir):
 
         assert ob2 == ob2_
 
+    def test_persisted_progress(self):
```

Comment:
missing test case for when the file did not exist at all. You should test it creating a default `PersistedProgress` and writing that default value to disk